tests/components/minecraft_server/test_config_flow.py

```diff
@@ -0,0 +1,194 @@
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_port"}
+
+
+async def test_port_too_large(hass: HomeAssistantType) -> None:
+    """Test error in case of a too large port."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_PORT_TOO_LARGE
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_port"}
+
+
+async def test_connection_failed(hass: HomeAssistantType) -> None:
+    """Test error in case of a failed connection."""
+    with patch("mcstatus.server.MinecraftServer.ping", side_effect=OSError):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT
+        )
+
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_connection_succeeded_with_host(hass: HomeAssistantType) -> None:
+    """Test config entry in case of a successful connection with a host name."""
+    with patch("mcstatus.server.MinecraftServer.ping", return_value=50):
+        with patch(
+            "mcstatus.server.MinecraftServer.status",
+            return_value=PingResponse(STATUS_RESPONSE_RAW),
+        ):
+            result = await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT
+            )
+
+            assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+            assert result["title"] == f"{USER_INPUT[CONF_HOST]}:{USER_INPUT[CONF_PORT]}"
+            assert result["data"][CONF_NAME] == USER_INPUT[CONF_NAME]
+            assert result["data"][CONF_HOST] == USER_INPUT[CONF_HOST]
+            assert result["data"][CONF_PORT] == USER_INPUT[CONF_PORT]
+
+
+async def test_connection_succeeded_with_ip4(hass: HomeAssistantType) -> None:
+    """Test config entry in case of a successful connection with an IPv4 address."""
+    with patch("getmac.get_mac_address", return_value="01:23:45:67:89:ab"):
+        with patch("mcstatus.server.MinecraftServer.ping", return_value=50):
+            with patch(
+                "mcstatus.server.MinecraftServer.status",
+                return_value=PingResponse(STATUS_RESPONSE_RAW),
+            ):
+                result = await hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_USER}, data=USER_INPUT_IPV4
+                )
+
+                assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+                assert (
+                    result["title"]
+                    == f"{USER_INPUT_IPV4[CONF_HOST]}:{USER_INPUT_IPV4[CONF_PORT]}"
+                )
+                assert result["data"][CONF_NAME] == USER_INPUT_IPV4[CONF_NAME]
+                assert result["data"][CONF_HOST] == USER_INPUT_IPV4[CONF_HOST]
+                assert result["data"][CONF_PORT] == USER_INPUT_IPV4[CONF_PORT]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4969180)